src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -102,6 +105,14 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (nric == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "NRIC"));
+        }
+        if (!Nric.isValidNric(nric)) {
+            throw new IllegalValueException(Nric.MESSAGE_CONSTRAINTS);
+        }
+        final Nric modelNric = nric;
```

Comment:
The construct for the Nric class is as such Nric(String value). So could you please change this line to 
final Nric modelNric = new Nric(nric)